cmd/manager/main.go

```diff
@@ -217,11 +217,14 @@ func newRootCommand() *cobra.Command {
 				log.Info("Starting the Cmd.")
 
 				// Start the Cmd
-				err = mgr.Start(signals.SetupSignalHandler())
+				stopCh := signals.SetupSignalHandler()
+				err = mgr.Start(stopCh)
 				if err != nil {
-					log.Fatal(err)
+					log.Error(err)
 				}
-				// Canceling the leader election context
+				// Stop controllers
+				stopCh <- struct{}{}
```

Comment:
This won't do anything. At this point, the controllers have already stopped. We need to send to the stop channel when the TERM signal is received in order to stop the controllers.

Reply:
That was already happening with the previous implementation I suppose. Look at the code [here](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/manager/signals/signal.go)

Reply:
I think this is taken care of by the original implementation of the `signals.SetupSignalHandler`

Reply:
Yeah, you are right. I steered you wrong here telling you to replace that call.

Reply:
The part that I was missing was that the either (1) the signals should not directly cause the release of the lock or (2) the release of the lock should block on the manager stopping.